velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -246,10 +262,12 @@ void PrestoHasher::hash<TypeKind::ARRAY>(
   auto rawNulls = baseArray->rawNulls();
   auto rawElementHashes = elementHashes->as<int64_t>();
   auto rawHashes = hashes->asMutable<int64_t>();
+  auto decodedNulls = vector_->nulls();
 
   rows.applyToSelected([&](auto row) {
     int64_t hash = 0;
-    if (!(rawNulls && bits::isBitNull(rawNulls, indices[row]))) {
+    if (!((decodedNulls && bits::isBitNull(decodedNulls, row)) ||
```

Comment:
we only need to check decodedNulls as DecodedVector collapses nulls from both the wrappings and the baseSe:
see: 
nulls pulled from wrapping: https://github.com/facebookincubator/velox/blob/main/velox/vector/DecodedVector.cpp#L145

nulls copied from inner dictionary layers:
https://github.com/facebookincubator/velox/blob/main/velox/vector/DecodedVector.cpp#L199-L226

nulls merged from flat base layer: https://github.com/facebookincubator/velox/blob/main/velox/vector/DecodedVector.cpp#L298